subsystems/sidecar/index.js

```diff
@@ -526,8 +526,28 @@ class Sidecar extends ReadyResource {
       return
     }
 
+    if (!hard && this.hasClients) {
+      const terminalClients = []
+      const seen = new Set()
+      for (const client of this.clients) {
+        const app = client.userData
+        if (!app || !app.state) continue // ignore e.g. `pear sidecar` cli i/o client
+        if (app?.state?.options?.type !== 'terminal') continue
+        if (seen.has(app.state.id)) continue
+        seen.add(app.state.id)
+
+        terminalClients.push(client)
+      }
+
+      if (this.verbose && terminalClients.length > 0) {
+        console.log(`Soft-restarting ${terminalClients.length} terminal app(s)`)
+      }
+      for (const client of terminalClients) client.userData.message({ type: 'pear/restart' })
+    }
+
```

Comment:
all of this should be no more than 5 LOC and just the one loop. do `const restarts = ... filter(...)` as a few lines below.

Reply:
also remove the log here - instead just put (hard) or (soft) in the console.log at the top of the restart function

Reply:
let's rename pear/restart to pear/reload and send this to all clients, not just terminal clients